File: linked.py
# 1
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
        
class LinkedList:
    def __init__(self):
        self.head = None

    def addLast(self, data):
        newNode = Node(data)

        if not self.head:
            self.head = newNode
            return

        current = self.head
        while current.next:
            current = current.next

        current.next = newNode

    def addFirst(self, data):
        self.head = Node(data, self.head)
    
    def removeFirst(self):
        if not self.head:
            return None
        current_head = self.head
        self.head = current_head.next
        return current_head.data

File: linked.js
// 1
class Node {
  constructor(data) {
    this.data = data;
    this.next = null;
  }
}

class LinkedList {
  constructor() {
    this.head = null;
  }

  addLast(data) {
    const newNode = new Node(data);

    if (!this.head) {
      this.head = newNode;
      return;
    }

    let current = this.head;
    while (current.next) {
      current = current.next;
    }

    current.next = newNode;
  }

  addFirst(data) {
		this.first = {
			data: data,
			next: this.first
		}
	}
	removeFirst() {
		let oldNode = this.first
		if(oldNode === null)
			return undefined
		this.first = oldNode.next
		return oldNode.data
	}
}

